src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -144,11 +144,23 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
       // Throw this exception so it can be handled at the reconnection layer, and don't show the
       // error notification.
       throw ex;
-    } else {
+    } else if (!isNestedError(ex)) {
       showErrorNotification();
+    } else if (ex.innerErrors && ex.innerErrors.some(e => e.errorType === 'UnsupportedTransportError' && e.transport === HttpTransportType.WebSockets)) {
+      logger.log(LogLevel.Error, 'Unable to connect, please ensure you are using an updated browser that supports WebSockets.');
+    } else if (ex.innerErrors && ex.innerErrors.some(e => e.errorType === 'FailedToStartTransportError' && e.transport === HttpTransportType.WebSockets)) {
+      logger.log(LogLevel.Error, 'Unable to connect, please ensure WebSockets are available. A VPN or proxy may be blocking the connection.');
+    } else if (ex.innerErrors && ex.innerErrors.some(e => e.errorType === 'DisabledTransportError' && e.transport === HttpTransportType.LongPolling)) {
+      logger.log(LogLevel.Error, 'Unable to initiate a SignalR connection to the server. This might be because the server is not configured to support WebSockets. To troubleshoot this, visit https://aka.ms/blazor-server-websockets-error.');
     }
   }
 
+  // Check if the connection is established using the long polling transport,
+  // using the `features.inherentKeepAlive` property only present with long polling.
+  if ((connection as any).connection.features.inherentKeepAlive) {
+    logger.log(LogLevel.Error, 'Failed to connect via WebSockets, using the Long Polling fallback transport. This may be due to a VPN or proxy blocking the connection. To troubleshoot this, visit https://aka.ms/blazor-server-using-fallback-long-polling.');
```

Comment:
https://aka.ms/blazor-server-using-fallback-long-polling will point to a new docs page which discusses common pitfalls and troubleshooting steps to ensure WebSockets are utilized (ie. enabled on IIS, enabled on azure portal, etc.)